gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java

```diff
@@ -688,14 +688,33 @@ public void exitParExpression(WUMLParser.ParExpressionContext ctx) {
         // For expressions with pattern "exp1" == "exp2"
         if (ctx.EQUAL() != null) {
             List<WUMLParser.ExpressionContext> expressions = ((WUMLParser.ParExpressionContext) ctx).expression();
-            // log.info(Integer.toString(expressions.size()));
-            // Works only for formats: "eval("$p1.p2")=="abc""
             if (expressions != null && expressions.size() == 2) {
-                String sourceDefinition = StringParserUtil
-                        .getValueWithinDoubleQuotes(expressions.get(0).evalExpression().StringLiteral().getText());
-                Source source = new Source(sourceDefinition);
-                String conditionValue = StringParserUtil
-                        .getValueWithinDoubleQuotes(expressions.get(1).literal().StringLiteral().getText());
+                String sourceDefinition, conditionValue;
+                Source source;
+                //TODO: A better way to do this?
+                if (expressions.get(0).evalExpression().getChild(2) instanceof TerminalNode) {
```

Comment:
In this case I think that is not possible because if we dont have 2nd child, deploying the integration-flow should itself fail since it doesn't comply with the grammar.
Anyway, I'm still working on this code. I will be walking through the code and improve.